src/lib/detection_algorithms/detectionUtils.js

```diff
@@ -0,0 +1,33 @@
+//random set http://stackoverflow.com/questions/2380019/generate-unique-random-numbers-between-1-and-100 there has to be a set way of doing this
+module.exports.generateTestIndecies = function(percent, taggedFramesCount, cb) {
+    var testIndecies = [];
+    while(testIndecies.length < Math.ceil(percent * taggedFramesCount)){
```

Comment:
Funny. this is an interview question that we used to ask.
Theoretically this code can run forever if you have a very crappy random number generator that will just generate the same numbers.
The idea is to do the random pick over a list of indices and then remove the chosen index on every iteration.
Take a look at this:
https://bost.ocks.org/mike/shuffle/